dowhy/causal_model.py

```diff
@@ -18,347 +19,378 @@
+				self._graph = CausalGraph(
+					self._treatment,
+					self._outcome,
+					common_cause_names=self._common_causes,
+					instrument_names=self._instruments,
+					effect_modifier_names=self._effect_modifiers,
+					observed_node_names=self._data.columns.tolist()
+				)
+			elif common_causes is not None:
+				self._graph = CausalGraph(
+					self._treatment,
+					self._outcome,
+					common_cause_names=self._common_causes,
+					effect_modifier_names = self._effect_modifiers,
+					observed_node_names=self._data.columns.tolist()
+				)
+			elif instruments is not None:
+				self._graph = CausalGraph(
+					self._treatment,
+					self._outcome,
+					instrument_names=self._instruments,
+					effect_modifier_names = self._effect_modifiers,
+					observed_node_names=self._data.columns.tolist()
+				)
+			else:
+				self._graph = None
+
+		else:
+			self.init_graph(graph=graph)
+			
+		self._other_variables = kwargs
+		self.summary()
+
+	def init_graph(self, graph):
+		'''
+		Initialize self._graph using graph provided by the user.
+
+		'''
+		# Create causal graph object
+		self._graph = CausalGraph(
+			self._treatment,
+			self._outcome,
+			graph,
+			effect_modifier_names=self._effect_modifiers,
+			observed_node_names=self._data.columns.tolist(),
+			missing_nodes_as_confounders = self._missing_nodes_as_confounders
+		)
+		self._common_causes = self._graph.get_common_causes(self._treatment, self._outcome)
+		self._instruments = self._graph.get_instruments(self._treatment,
+														self._outcome)
+		# Sometimes, effect modifiers from the graph may not match those provided by the user.
+		# (Because some effect modifiers may also be common causes)
+		# In such cases, the user-provided modifiers are used.
+		# If no effect modifiers are provided,  then the ones from the graph are used.
+		if self._effect_modifiers is None or not self._effect_modifiers:
+			self._effect_modifiers = self._graph.get_effect_modifiers(self._treatment, self._outcome)
+
+	def learn_graph(self, method_name="cdt.causality.graph.LiNGAM", render=False, view=False, *args, **kwargs):
+		'''
+		Learn causal graph from the data.
+		'''
+
+		if method_name is None:
+			#TODO add LiNGAM as default backdoor method, DirectLiNGAM as default LiNGAM method, add an informational message to show which method has been selected.
+			pass
+		else:
+			library_class = graph_learners.get_library_class_object(method_name)
+			str_arr = method_name.split(".", maxsplit=1)
+			library_name = str_arr[0]
+			# discovery_name = str_arr[1]
+			
+			causal_discovery_class = graph_learners.get_discovery_class_object(library_name)
+	
+		model = causal_discovery_class(self._data, library_class, *args, **kwargs)
+		graph = model.learn_graph()
+		
+		# Initialize causal graph object
+		self.init_graph(graph=graph)
+		
+		# Save and/or view the graph
+		if render:
+			model.render("image.png", view=view)
+
+	def identify_effect(self, estimand_type=None,
```

Comment:
is there any change to this method? not sure why github diffshows it as changed.

Reply:
Which line specifically? I cant seem to find any in other functions.